vulkaninfo/vulkaninfo.cpp

```diff
@@ -355,8 +354,8 @@ void GpuDumpProps(Printer &p, AppGpu &gpu) {
     p.AddNewline();
 }
 void GpuDumpQueueProps(Printer &p, std::vector<SurfaceExtension> &surfaces, AppQueueFamilyProperties &queue) {
-    p.SetElementIndex(static_cast<int>(queue.queue_index)).SetSubHeader().ObjectStart("queueProperties");
-    if (p.Type() == OutputType::json) {
+    p.SetSubHeader().SetElementIndex(static_cast<int>(queue.queue_index)).ObjectStart("queueProperties");
+    if (p.Type() == OutputType::json || p.Type() == OutputType::vkconfig_output) {
```

Comment:
Seems like this if-else tree (and the two below) would be more clearly expressed by switch()es.

(Although so many conditionals based on p.Type() throughout the code really suggest a refactoring into subclasses)

Reply:
Most of these if-else nests were just from json schema handling. I moved all the json dumping logic into their own functions and make the entire codebase much simpler to read